src/software/util/parameter/generate_parameters_v2.py

```diff
@@ -0,0 +1,359 @@
+
+            # the config name is the name of the configuration class
+            config_name=self.config_name,
+
+            # the entries that will go into the constructor of the config
+            # class excluding the parameter_list initialization
+            constructor_entries=constructor_config_entires + constructor_parameter_entries,
+
+            # parameter list initialization with all the parameters and configs
+            # that are a part of this class, without the const pointer cast
+            mutable_parameter_list_entries=',\n'.join(
+                mutable_parameter_list_entries),
+
+            # parameter list initialization with all the parameters and configs
+            # that are a part of this class, with the const pointer cast
+            immutable_parameter_list_entries=',\n'.join(
+                immutable_parameter_list_entries),
+
+            # public members of the config class
+            public_entries=public_parameter_entries + public_config_entries,
+
+            # private members of the config class
+            private_entries=private_parameter_entries + private_config_entries
+        )
+
+    def get_constructor_entries(self):
+        """Returns the c++ code needed in the constructor of the parent config
+        for this child config to be nested in.
+
+        :returns: entry to put in constructor of ANOTHER config
+        :rtype: str
+
+        """
+        return constants.CONFIG_CONSTRUCTOR_ENTRY.format(
+            config_name=self.config_name,
+            config_variable_name=self.config_variable_name)
+
+    def get_private_entries(self):
+        """Returns the private members of the parent Config class that are required
+        for this nested Config.
+
+        :returns: entries into the private section of the Config class
+        :rtype: str
+
+        """
+        return constants.CONFIG_PRIVATE_ENTRY.format(
+            config_name=self.config_name,
+            config_variable_name=self.config_variable_name)
+
+    def get_public_entries(self):
+        """Returns the public members of the parent Config class that are required
+        for this nested Config.
+
+        :returns: entries into the public section of the Config class
+        :rtype: str
+
+        """
+        return constants.CONFIG_PUBLIC_ENTRY.format(
+            config_name=self.config_name,
+            immutable_accessor_name='get'+self.config_name.strip(),
+            mutable_accessor_name='getMutable'+self.config_name.strip(),
+            config_variable_name=self.config_variable_name)
+
+
+#######################################################################
+#                                MAIN                                 #
+#######################################################################
+if __name__ == '__main__':
+
+    # create config
+    ThunderbotsConfig =\
+        Config('ThunderbotsConfig', load_configuration(sys.argv[1:-1]))
```

Comment:
Are we just loading one config? Or what is this top-level name for?

Reply:
all the yaml files are loaded into this one config, so you could access all the top level configs from this one config